### PR TITLE
MAID-2078 Mock crypto

### PR DESCRIPTION
**Pull request by madadam**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: .travis.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ script:
         cargo rustc --verbose --example ci_test -- -Zno-trans &&
         cargo rustc --verbose --example key_value_store -- -Zno-trans &&
         cargo rustc --verbose -- -Zno-trans &&
-        travis_wait 60 cargo test --release --verbose --features use-mock-crust
+        travis_wait 60 cargo test --release --verbose --features \"use-mock-crust use-mock-crypto\"
```
Comment (afck) [inline .travis.yml]: I think we can remove this `travis_wait 60` now.

File: src/data/immutable_data.rs [hunk 2]
```diff
@@ -87,12 +87,14 @@ impl Debug for ImmutableData {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use hex::ToHex;
     use maidsafe_utilities::{SeededRng, serialisation};
     use rand::Rng;
 
     #[test]
+    #[cfg(not(feature = "use-mock-crypto"))]
```
Comment (afck) [inline src/data/immutable_data.rs]: That test should work with or without mock crypto, shouldn't it? It's just the value of `expected_name` below that needs to be conditionally compiled.

File: src/mock_crypto.rs [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+//! Mock cryptographic primitives.
+
+/// Mock version of a subset of the rust_sodium crate.
```
Comment (afck) [inline src/mock_crypto.rs]: Maybe explain that these are not secure but make the tests several times faster.

File: src/mock_crypto.rs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+            use rand::Rng;
+
+            /// Number of bytes in a 'PublicKey`.
+            pub const PUBLICKEYBYTES: usize = 32;
+            /// Number of bytes in a `SecretKey`.
+            pub const SECRETKEYBYTES: usize = 32;
+            /// Number of bytes in a `Nonce`.
+            pub const NONCEBYTES: usize = 4;
+
+            /// Mock public key for asymetric encryption/decryption.
```
Comment (afck) [inline src/mock_crypto.rs]: `asymmetric` (also below)

**Comment by afck** [comment]
My comment got hidden: we can remove the `travis_wait 60` now.

**Pull request merged**